io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,356 @@
+    {
+      auto clr = getTextureColor ( texture, uvptr->u, uvptr->v );
+
+      p.r = clr.r;
+      p.g = clr.g;
+      p.b = clr.b;
+    } );
+  }
+
+  pcl::PointCloud<pcl::PointXYZRGBA>::Ptr
+  RealSense2Grabber::convertRGBADepthToPointXYZRGBA ( const rs2::points& points, const rs2::video_frame& texture )
+  {
+    return convertRealsensePointsToPointCloud<pcl::PointXYZRGBA> ( points, [&]( pcl::PointXYZRGBA& p, const rs2::texture_coordinate* uvptr )
+    {
+      auto clr = getTextureColor ( texture, uvptr->u, uvptr->v );
+
+      p.r = clr.r;
+      p.g = clr.g;
+      p.b = clr.b;
+    } );
+  }
+
+  pcl::PointCloud<pcl::PointXYZI>::Ptr
+  RealSense2Grabber::convertIntensityDepthToPointXYZRGBI ( const rs2::points& points, const rs2::video_frame& texture )
+  {
+    if (texture.get_profile ().format () == RS2_FORMAT_UYVY)
+    {
+      return convertRealsensePointsToPointCloud<pcl::PointXYZI> ( points, [&]( pcl::PointXYZI& p, const rs2::texture_coordinate* uvptr )
+      {
+        auto clr = getTextureColor ( texture, uvptr->u, uvptr->v );
+        p.intensity = 0.299f * static_cast <float> (clr.r) + 0.587f * static_cast <float> (clr.g) + 0.114f * static_cast <float> (clr.b);
+      } );
+    }
+    else
+    {
+      return convertRealsensePointsToPointCloud<pcl::PointXYZI> ( points, [&]( pcl::PointXYZI& p, const rs2::texture_coordinate* uvptr )
+      {
+        p.intensity = getTextureIntensity ( texture, uvptr->u, uvptr->v );
+      } );
+    }
+  }
+
+  template <typename PointT, typename Functor>
+  typename pcl::PointCloud<PointT>::Ptr
+  RealSense2Grabber::convertRealsensePointsToPointCloud ( const rs2::points& points, Functor mapColorFunc )
+  {
+    typename pcl::PointCloud<PointT>::Ptr cloud ( new pcl::PointCloud<PointT> () );
+
+    auto sp = points.get_profile ().as<rs2::video_stream_profile> ();
+    cloud->width = sp.width ();
+    cloud->height = sp.height ();
+    cloud->is_dense = false;
+    cloud->points.resize ( points.size () );
+
+    const auto cloud_vertices_ptr = points.get_vertices ();
+    const auto cloud_texture_ptr = points.get_texture_coordinates ();
+
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
+    {
+      const auto ptr = cloud_vertices_ptr + index;
+      const auto uvptr = cloud_texture_ptr + index;
+      auto& p = cloud->points[index];
+
+      p.x = ptr->x;
+      p.y = ptr->y;
+      p.z = ptr->z;
+
+      mapColorFunc ( p, uvptr );
+    }
+
+    return cloud;
+  }
+
+  pcl::RGB
+  RealSense2Grabber::getTextureColor ( const rs2::video_frame& texture, float u, float v )
+  {
+    const int w = texture.get_width (), h = texture.get_height ();
+    int x = std::min ( std::max ( int ( u*w + .5f ), 0 ), w - 1 );
+    int y = std::min ( std::max ( int ( v*h + .5f ), 0 ), h - 1 );
+    int idx = x * texture.get_bytes_per_pixel () + y * texture.get_stride_in_bytes ();
```

Comment:
Proposal to define a new static protected method called `getTextureIdx`. It includes the 4 lines abose and returns an index. The type should be `size_t` instead of `int` or am I missing something.